src/core/lib/security/security_connector/tls/tls_security_connector.cc

```diff
@@ -334,14 +334,12 @@ grpc_security_status TlsChannelSecurityConnector::InitializeHandshakerFactory(
       static_cast<const TlsCredentials*>(channel_creds());
   grpc_tls_key_materials_config* key_materials_config =
       creds->options().key_materials_config();
-  /* Copy key materials config from credential options. */
+  // key_materials_config_->set_key_materials will handle the copying of the key
```

Comment:
It seems only copying `pem_root_certs` but not  `pem_key_cert_pair_list`.

Reply:
In this comment, I meant the copy was handled inside key_materials_config_->set_key_materials. So in this layer we could just pass the pointer to the function, and it will be copied inside the function.